src/Symfony/Component/DependencyInjection/Compiler/ResolveBindingsPass.php

```diff
@@ -37,11 +38,19 @@ public function process(ContainerBuilder $container)
             parent::process($container);
 
             foreach ($this->unusedBindings as list($key, $serviceId)) {
-                throw new InvalidArgumentException(sprintf('Unused binding "%s" in service "%s".', $key, $serviceId));
+                $message = sprintf('Unused binding "%s" in service "%s".', $key, $serviceId);
+                if ($this->errors) {
+                    $message .= sprintf("\nCould be related to%s:", 1 < \count($errors) ? ' one of' : '');
+                }
+                foreach ($this->errors as $serviceId => $mesage) {
```

Comment:
Not sure if the missing `s` is voluntary here, but we can't use `$message` since it's used in the parent scope, maybe `$error` ?